src/neo/SmartContract/Native/StdLib.cs

```diff
@@ -135,9 +137,48 @@ public static string Base58Encode(byte[] data)
         /// <param name="s">The base58 <see cref="string"/>.</param>
         /// <returns>The decoded byte array.</returns>
         [ContractMethod(CpuFee = 1 << 12)]
-        public static byte[] Base58Decode(string s)
+        public static byte[] Base58Decode([MaxLength(MaxInputLength)] string s)
         {
             return Base58.Decode(s);
         }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemoryCompare([MaxLength(MaxInputLength)] byte[] str1, [MaxLength(MaxInputLength)] byte[] str2)
+        {
+            return Math.Sign(str1.AsSpan().SequenceCompareTo(str2));
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemorySearch(byte[] mem, byte[] value)
+        {
+            return MemorySearch(mem, value, 0, false);
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemorySearch(byte[] mem, byte[] value, int start)
+        {
+            return MemorySearch(mem, value, start, false);
```

Comment:
I think that we will need the attribute in both methods, otherwise it won't call `Validate`
```suggestion
            return MemorySearch([MaxLength(MaxInputLength)] mem, value, start, false);
```